components/brave_wallet/browser/zcash/zcash_wallet_service.cc

```diff
@@ -374,11 +373,15 @@ void ZCashWalletService::SignAndPostTransaction(
     const mojom::AccountIdPtr& account_id,
     const ZCashTransaction& zcash_transaction,
     SignAndPostTransactionCallback callback) {
-  complete_manager_.CompleteTransaction(
-      chain_id, zcash_transaction, account_id.Clone(),
-      base::BindOnce(&ZCashWalletService::CompleteTransactionDone,
-                     weak_ptr_factory_.GetWeakPtr(), chain_id,
-                     zcash_transaction, std::move(callback)));
+  auto& task = complete_transaction_tasks_.emplace_back(
+      std::make_unique<ZCashCompleteTransactionTask>(
+          base::PassKey<class ZCashWalletService>(), *this,
```

Comment:
What's the expected lifetime on the `*this` here? It seems like it's meant to match the weak_ptr_factory, but I just want to make sure.

Reply:
ZCashCompleteTransactionTask object is being kept as a part of **this** object so it shouldn't over-live ***this**

Reply:
Cool makes sense - no concerns